src/cosmos/cosmosUtils.ts

```diff
@@ -210,3 +202,20 @@ export const reduceCoinEventsForAddress = (
   })
   return out
 }
+
+// The helper function parseCoins from the @cosmjs sdk doesn't handle denoms with hyphens.
+const extendedParseCoins = (input: string): Coin[] => {
```

Comment:
This needs some unit tests. Also testing this with `usara-core1r9gc0rnxnzpq33u82f44aufgdwvyxv4wyepyck98m9v2pxua6naqr8h03z` causes the routine to throw. 

Should this work or is `amount` not the denom.

Reply:
This parses amount/denom strings from transactions ie. 12345usara-core1r9gc0rnxnzpq33u82f44aufgdwvyxv4wyepyck98m9v2pxua6naqr8h03z